include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1008 @@
+                        }
+                        outFile << std::endl;
+                        fisttimestep = false;
+                    } 
+                    outFile << currentStep << " "
+                            << std::scientific;
+
+                    long double numElec_all = 0.0;
+                    long double ux2_all = 0.0;
+                    long double pos2_SI_all = 0.0;
+                    long double xux_all = 0.0;
+
+                    for (int i = startWindow_y; i < endWindow_y; i++)
+                    {
+                            numElec_all+= static_cast<long double>(globalCount_e.getDataPointer( )[i]);
+                            ux2_all    += static_cast<long double>(globalSumMom2.getDataPointer( )[i]) * UNIT_MASS* UNIT_MASS /(SI::ELECTRON_MASS_SI*SI::ELECTRON_MASS_SI);
+                            pos2_SI_all+= static_cast<long double>(globalSumPos2.getDataPointer( )[i]) *UNIT_LENGTH*UNIT_LENGTH ;
+                            xux_all    += static_cast<long double>(globalSumMomPos2.getDataPointer( )[i])*UNIT_MASS *UNIT_LENGTH /SI::ELECTRON_MASS_SI;
+                    }
+                    double emit_all = algorithms::math::sqrt(((double)pos2_SI_all * (double)ux2_all - (double)xux_all * (double)xux_all)/((double)numElec_all*(double)numElec_all));
+                    if (emit_all > 0.0){
+                        outFile << emit_all << " ";
+                    }
+                    else {
+                        outFile << "0.0 ";
+                    }
+
+                    for (int i = startWindow_y; i < endWindow_y; i+=10)
+                    {
+                        double numElec   = globalCount_e.getDataPointer( )[i];
+                        double mom2_SI   = globalSumMom2.getDataPointer( )[i] * UNIT_MASS * UNIT_SPEED* UNIT_MASS * UNIT_SPEED;
+                        double pos2_SI   = globalSumPos2.getDataPointer( )[i] *UNIT_LENGTH*UNIT_LENGTH ;
+                        double mompos_SI = globalSumMomPos2.getDataPointer( )[i]*UNIT_MASS * UNIT_SPEED*UNIT_LENGTH;
+                        for ( int j=i+1; j<i+10 && j < endWindow_y; j++ ){
+                                numElec   += globalCount_e.getDataPointer( )[j];
+                                mom2_SI   += globalSumMom2.getDataPointer( )[j] * UNIT_MASS * UNIT_SPEED* UNIT_MASS * UNIT_SPEED;
+                                pos2_SI   += globalSumPos2.getDataPointer( )[j] *UNIT_LENGTH*UNIT_LENGTH;
+                                mompos_SI += globalSumMomPos2.getDataPointer( )[j]*UNIT_MASS * UNIT_SPEED*UNIT_LENGTH;
+                        }
+                        double ux2 = mom2_SI/(UNIT_SPEED*UNIT_SPEED*SI::ELECTRON_MASS_SI*SI::ELECTRON_MASS_SI);
+                        double xux = mompos_SI/(UNIT_SPEED*SI::ELECTRON_MASS_SI);
+                        double emit = algorithms::math::sqrt((pos2_SI * ux2 - xux * xux)/ numElec/ numElec);
+                        if(numElec<std::numeric_limits<float>::epsilon()){
```

Comment:
do not use `float`. If this value has a variable precision than use `float_X` but it looks like it must fit with the calculated values and must be `float_64`